### PR TITLE
Pub/sub enabled and checks

### PR DESCRIPTION
**Pull request by johanstokking**
#### Summary

Groundwork for https://github.com/TheThingsIndustries/lorawan-stack/issues/2442

#### Changes

- Allow pub/sub providers to be disabled
- Open pub/sub provider with enabled check
- Connect explicitly to invalid server

#### Testing

Unit testing and local testing with NATS

##### Regressions

None expected

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by adriansmares — APPROVED** [review]
:+1:

You may want pull the `allEnabled` into `io/pubsub/provider` as an `util_test.go` probably.

File: pkg/applicationserver/io/pubsub/pubsub.go [hunk 1]
```diff
@@ -218,6 +227,18 @@ func (i *integration) startHandleDown(ctx context.Context) {
 	}
 }
 
+// OpenConnection implements provider.Opener.
+func (ps *PubSub) OpenConnection(ctx context.Context, target provider.Target, enabler provider.Enabler) (*provider.Connection, error) {
+	provider, err := provider.GetProvider(target)
+	if err != nil {
+		return nil, err
+	}
+	if err := ps.providerStatuses.Enabled(ctx, target.GetProvider()); err != nil {
```
Comment (adriansmares) [inline pkg/applicationserver/io/pubsub/pubsub.go]: I think this should be `enabler.Enabled` for consistency. Or you may drop the `OpenConnection` method completely.

**Review by htdvisser — APPROVED** [review]
Changes to default config look OK. IIRC those were the defaults anyway, right?

Did not review `pkg/applicationserver`

**Comment by johanstokking** [comment]
> Changes to default config look OK. IIRC those were the defaults anyway, right?

Yes, this comes unchanged from TTSE and does not affect behavior of TTS.

**Pull request merged**